### PR TITLE
Failure during preview with Max Rows filter and 2+ source tables

### LINKED ISSUE #706: Previews fail when multiple tables present in job
If there is more than one source table available in a job, previews stops working, because of the max rows filter:
```
java.lang.IllegalStateException: Component has input columns from multiple tables: ImmutableFilterJob[name=null,filter=Max rows]
    java.util.concurrent.FutureTask.report(FutureTask.java:122)
    java.util.concurrent.FutureTask.get(FutureTask.java:192)
    javax.swing.SwingWorker.get(SwingWorker.java:602)
    org.datacleaner.windows.DataSetWindow$2.done(DataSetWindow.java:149)
    javax.swing.SwingWorker$5.run(SwingWorker.java:737)
    javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(SwingWorker.java:832)
    sun.swing.AccumulativeRunnable.run(AccumulativeRunnable.java:112)
    javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(SwingWorker.java:842)
    javax.swing.Timer.fireActionPerformed(Timer.java:313)
    javax.swing.Timer$DoPostEvent.run(Timer.java:245)
    java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
    java.awt.EventQueue.dispatchEventImpl(EventQueue.java:744)
    java.awt.EventQueue.access$400(EventQueue.java:97)
    java.awt.EventQueue$3.run(EventQueue.java:697)
    java.awt.EventQueue$3.run(EventQueue.java:691)
    java.security.AccessController.doPrivileged(Native Method)
    java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
    java.awt.EventQueue.dispatchEvent(EventQueue.java:714)
    java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
    java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
    java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Component has input columns from multiple tables: ImmutableFilterJob[name=null,filter=Max rows]
    org.datacleaner.job.runner.RowProcessingPublishers.getTables(RowProcessingPublishers.java:219)
    org.datacleaner.job.runner.RowProcessingPublishers.registerRowProcessingPublishers(RowProcessingPublishers.java:233)
    org.datacleaner.job.runner.RowProcessingPublishers.registerJob(RowProcessingPublishers.java:143)
    org.datacleaner.job.runner.RowProcessingPublishers.registerAll(RowProcessingPublishers.java:126)
    org.datacleaner.job.runner.RowProcessingPublishers.<init>(RowProcessingPublishers.java:122)
    org.datacleaner.job.runner.AnalysisRunnerJobDelegate.run(AnalysisRunnerJobDelegate.java:103)
    org.datacleaner.job.runner.AnalysisRunnerImpl.run(AnalysisRunnerImpl.java:90)
    org.datacleaner.actions.PreviewTransformedDataActionListener.call(PreviewTransformedDataActionListener.java:134)
    org.datacleaner.actions.PreviewTransformedDataActionListener.call(PreviewTransformedDataActionListener.java:52)
    org.datacleaner.windows.DataSetWindow$2.doInBackground(DataSetWindow.java:144)
    org.datacleaner.windows.DataSetWindow$2.doInBackground(DataSetWindow.java:142)
    javax.swing.SwingWorker$1.call(SwingWorker.java:295)
    java.util.concurrent.FutureTask.run(FutureTask.java:266)
    javax.swing.SwingWorker.run(SwingWorker.java:334)
    java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    java.lang.Thread.run(Thread.java:745)
```

### PR DESCRIPTION
**Pull request by kaspersorensen**
Fixes #706 

This turned out to "just" be a matter of only focusing on a single source table. But while testing I also saw that the Max Rows filter did not have any direct association with that source table and that it did not have a name, so it would easily be "equal" to other max rows filters in the job. Therefore the generated preview-job now has those things added.

### CODE REVIEW

**Comment by kaspersorensen** [comment]
Just saw the build fail and here's a few notes for self:

 * I think what is going wrong here is that the query optimizer is applying BOTH max rows filters as an optimization so they overwrite each other.

**Comment by kaspersorensen** [comment]
OK, should be mergeable now. But I will immediately go look at
https://issues.apache.org/jira/browse/METAMODEL-194

File: desktop/ui/src/test/java/org/datacleaner/actions/PreviewTransformedDataActionListenerTest.java [hunk 1]
```diff
@@ -70,6 +72,19 @@ protected void setUp() throws Exception {
         emailTransformerBuilder.addInputColumn(analysisJobBuilder.getSourceColumnByName("EMAIL"));
     }
 
+    public void testJobWithMaxRowsFilter() throws Exception {
+        FilterComponentBuilder<MaxRowsFilter, Category> filter = analysisJobBuilder.addFilter(MaxRowsFilter.class);
+        filter.addInputColumn(analysisJobBuilder.getSourceColumnByName("email"));
```
Comment (ClaudiaPHI) [inline desktop/ui/src/test/java/org/datacleaner/actions/PreviewTransformedDataActionListenerTest.java]: some finals here and there.

Comment (kaspersorensen) [inline desktop/ui/src/test/java/org/datacleaner/actions/PreviewTransformedDataActionListenerTest.java]: Added finals, but this line didn't need any :)

**Comment by kaspersorensen** [comment]
Otherwise OK @ClaudiaPHI ?

**Comment by ClaudiaPHI** [comment]
LGTM!

**Pull request merged**